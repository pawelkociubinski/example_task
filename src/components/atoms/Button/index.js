/* @flow */
// Dependencies
import type { Node } from 'react';
import React from 'react';
import { noop } from 'lodash';
import { ThemeProvider } from 'styled-components';
// Custom
import themes, { CircleButton, RectangularButton } from './styles';

const ButtonComponents = {
  circle: CircleButton,
  rectangle: RectangularButton,
};

type Props = {
  children: Node,
  handleClick?: () => void,
  shape: string,
  theme: string,
}

const Button = (props: Props) => {
  const {
    children, shape, handleClick, theme,
  } = props;

  const ButtonComponent = ButtonComponents[shape];
  return (
    <ThemeProvider theme={themes[theme]}>
      <ButtonComponent onClick={handleClick}>
        {children}
      </ButtonComponent>
    </ThemeProvider>
  );
};

Button.defaultProps = {
  shape: 'rectangle',
  theme: 'primary',
  handleClick: noop,
};

export default Button;
