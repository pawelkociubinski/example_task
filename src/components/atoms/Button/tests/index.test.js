import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('<Button />', () => {
  it('renders correctly', () => {
    const onClickMocked = jest.fn();
    const component = renderer.create(<Button handleClick={onClickMocked}>Button</Button>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
