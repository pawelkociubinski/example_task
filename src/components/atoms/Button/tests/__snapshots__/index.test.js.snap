// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #DDDDDD;
  color: #707070;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  overflow: hidden;
  position: relative;
  border: none;
  padding: 0 40px;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  Button
</button>
`;
